### ISSUE TITLE
firewall test strange results

### ISSUE THREAD

**Issue by bol-van** [PurpleI2P/i2pd#1845]
Версия 2.45  openwrt x86-64,, native ipv4 + ipv6

Что-то совсем плохо стал работать тест сети.
После перезапуска рандомная картина по v4 и v6.

Варианты 

1) Почти сразу OK
2) Долго testing , потом unknown
3) Быстро firewalled

порты открыты, но кто знает может где-то что-то в сети заблокировано, и с каких-то пиров нет доступа
как реализован тест открытости порта ?

**Comment by bol-van** [comment]
На VPSке 2.45 работает нормально. Всегда "OK".

Пробовал переносить слинкованный с musl  i2pd на vps и с vps линкованный с glibc на openwrt. Обе версии ведут себя почти одинаково.

**Comment by Vort** [comment]
Стоит потестировать вот этот коммит a3c305032a5490a1815f3f60a0f21c4ddeed7c10.
Есть шанс, что на этой версии пиртест станет работать лучше.

**Comment by orignal** [comment]
Так может и правда какие то проблемы с сетью?

**Comment by bol-van** [comment]
В чем они могут заключаться ? Как это проверить ? Может в логах что-то в debug режиме искать ?

Есть мысль перезагрузиться в дебик и на нем попробовать, но я не там нахожусь. Удаленно только отрежу себя от системы, сделав что-то не то.

**Comment by bol-van** [comment]
Что вообще такое состояние Unknown ?
Testing ... 1 секунда ... Unknown
как это могут вызвать проблемы с сетью ?

**Comment by orignal** [comment]
Unknown это значит тест заврешился с ошибкой и невозможно понять что на самом деле.
Проблемы с сетью это если Firewalled.

**Comment by bol-van** [comment]
firewalled довольно редко, гораздо чаще unknown

**Comment by orignal** [comment]
Unknown это на данный момент не проблема. В целом работаем над улучшением.

**Comment by bol-van** [comment]
Теперь на VPSке, где раньше было все ок,  стабильно по ipv4 Testing
сколько ни перезапускай
порт точно открыт, телнет подключается.
количество роутеров упало как обычно

**Comment by orignal** [comment]
>стабильно по ipv4 Testing
раз Testing значит не приходят ответы по UDP

>порт точно открыт, телнет подключается.
Так это TCP, а не UDP

**Comment by bol-van** [comment]
он не слушает на udp ipv4, вот в чем проблема
Есть только NTCP2, NTCP2v6, SSU2v6

на порту udp нет других листенеров
если повесить неткат на удп, подключиться с другого нетката, то все приходит

**Comment by Vort** [comment]
> он не слушает на udp ipv4, вот в чем проблема

Может, в логах будет подсказка?

**Comment by bol-van** [comment]
В логах ничего интересного не нашел
Игры с параметрами address4 и ifname не влияют ни на что

**Comment by Vort** [comment]
Ещё можно попробовать удалить/переименовать `router.info`.
Помню когда-то у меня были проблемы с его заклиниванием.

**Comment by bol-van** [comment]
Jan 22 17:05:07 vps daemon.info i2pd[483791]: [541] NTCP2: Start listening v4 TCP port 7819
Jan 22 17:05:07 vps daemon.info i2pd[483791]: [541] NTCP2: Start listening v6 TCP port 7819
Jan 22 17:05:07 vps daemon.info i2pd[483791]: [541] SSU2: Start listening on [...ipv6...]:7819

он даже не пытается

**Comment by bol-van** [comment]
Если посмотреть в hex что в router.info, то как раз вижу 3 записи
NTCP2, NTCP2 host=ipv6, SSU2 host=ipv6
у здорового роутера там 4 записи

вот и причина. портится router.info

**Comment by Vort** [comment]
так всё же - если убрать порченный, создастся ли нормальный?

**Comment by bol-van** [comment]
Да, исправляется при удалении router.info.  router.keys не удалял

**Comment by Vort** [comment]
Заклинивания router.info - это, похоже, популярная проблема в i2pd.
Надо бы проверить, задокументирована ли она тут в Issues.
Если нет, то стоит сделать.

Только я так и не выяснил, при каких условиях это происходит.
Шаги же для воспроизведения найти было бы очень желательно.

**Comment by bol-van** [comment]
Я бы начала с разборка где и когда этот файл вообще записывается. И в чем необходимость его регулярной перезаписи

**Comment by orignal** [comment]
То что поломался router.info из-за изменения конфига это старая проблема. Нужен точный сценарий когда это происходит.
